BZ1850510 - remove NFS from image reg docs

bobfuru opened the pull request:
This work updates content that was added in https://github.com/openshift/openshift-docs/pull/24276/ and is a continuation of [BZ 1850510](https://bugzilla.redhat.com/show_bug.cgi?id=1850510). Based on consultation with @huffmanca, this PR:

- Removes NFS references throughout bare metal and vsphere registry storage docs.
- Updates the notes that begin with, "When using shared storage..." and reworded these to say, "When using shared storage review your security settings to prevent outside access".
- Updates the procedures that previously had references to NFS and RWX. For vSphere, I used the PV/PVC examples from https://docs.openshift.com/container-platform/4.4/storage/persistent_storage/persistent-storage-vsphere.html
- Removes xrefs to NFS Group IDs topic.
- Updates the prereqs note about provisioned storage in both vSphere and bare metal to indicate new functionality introduced in 4.4 that supports RWO if you have only one replica. Hence, this PR will be CP'ed only back to 4.4.

**Preview links:**
- [Configuring registry storage for VMware vSphere](https://bz1850510-update--ocpdocs.netlify.app/openshift-enterprise/latest/registry/configuring_registry_storage/configuring-registry-storage-vsphere.html#registry-configuring-storage-vsphere_configuring-registry-storage-vsphere)
- [Configuring registry storage for bare metal](https://bz1850510-update--ocpdocs.netlify.app/openshift-enterprise/latest/registry/configuring_registry_storage/configuring-registry-storage-baremetal.html#registry-configuring-storage-baremetal_configuring-registry-storage-baremetal)

bobfuru commented:
@huffmanca PTAL, thank you!

bobfuru commented:
@qinpingli PTAL for QE, especially in the PV/PVC examples that were updated, thanks!

qinpingli commented on modules/registry-configuring-storage-baremetal.adoc:
```diff
@@ -16,9 +16,13 @@ registry to use storage.
+[IMPORTANT]
+====
+{product-title} supports `ReadWriteOnce` access mode for image registry storage when you have only one replica. To deploy a private image registry, or to configure registry storage for high availability with two or more replicas, you must use a different storage back end that supports `ReadWriteMany`, such as object storage.
+====
```
I don't know if we support object storage in baremetal cluster?
@wzheng1 Did we test object storage as backend storage for image registry?

wzheng1 replied:
Image registry is recommended to use object storage (S3, GCS, Swift, etc.), but I don't know what object storage can be used in a baremetal cluster?

wzheng1 replied:
I suggest to remove these statement, since they are no related to Baremetal and vSphere configuration: to configure registry storage for high availability with two or more replicas, you must use a different storage back end that supports `ReadWriteMany`, such as object storage.

bobfuru replied:
^ This statement was included as a rewording of this statement by @dmage:

> Image registry storage requires ReadWriteMany only for HA configuration (2+ replicas). Since 4.4 there is an option to use RWO if you have only one replica.

So perhaps we could leave the proposed statement as is for vSphere, and drop "such as object storage" for bare metal?

wzheng1 replied:
One more thing is about "To deploy a private image registry", maybe it should be To deploy an image registry which supports HA, ReadWriteMany is needed.

bobfuru replied:
Resolved

qinpingli commented on modules/registry-configuring-storage-baremetal.adoc:
```diff
@@ -40,30 +44,7 @@ $ oc get pod -n openshift-image-registry
-----
-exporting *:/mnt/data
-----
+If the storage type is `emptyDIR`, the replica number cannot be greater than `1`.
```
Seems for the `image-registry` deployment created by image registry cluster operator doesn't support `supplementalGroups`, so:
1. By default, if we use nfs storage as image registry backend storage, the replica number cannot be greater than `1`.
2. If the image registry cluster operator is unmanaged by CVO, and the `image-registry` deployment has the `supplementalGroups` config, the replica number can be greater than `1` for nfs storage.

wzheng1 replied:
If use emptyDir, the replica number of image registry has to be 1. If not, it will have synchronization issue.

bobfuru replied:
@qinpingli The purpose of this change is to delete #1 because it references `nfs`. We want to remove references to `nfs` in bare metal and vSphere image registry config docs. So we should be fine leaving the `emptyDIR` note, IIUC.

qinpingli replied:
@bobfuru Yes, agree with keeping `emptyDIR` note. If we need to add some notes about `NFS`?

bobfuru replied:
@qinpingli For `modules/registry-configuring-storage-baremetal.adoc` and `modules/registry-configuring-storage-vsphere.adoc`, we are removing notes about `NFS` based on the findings by @huffmanca that "NFS can't be a recommended way of using image registry storage, as you would either have to leave the deployment/operator in an unmanaged state (not recommended) or leave the NFS export exposed (world readable/writable)."

qinpingli commented on modules/registry-configuring-storage-vsphere.adoc:
```diff
@@ -10,35 +10,31 @@
 ====
-vSphere volumes do not support the `ReadWriteMany` access mode. You must use
-a different storage backend, such as object storage, to configure the registry storage for high-availability.
+{product-title} supports `ReadWriteOnce` access mode for image registry storage when you have only one replica. To deploy a private image registry, or to configure registry storage for high availability with two or more replicas, you must use a different storage back end that supports `ReadWriteMany`, such as object storage.
```
Didn't if we tested object storage in vSphere either.

bobfuru replied:
We have a similar note already in 4.4, 4.5: 

> vSphere volumes do not support the ReadWriteMany access mode. You must use a different storage backend, **_such as object storage_**, to configure the registry storage for high-availability.

So is it accurate still to use the wording "such as object storage", or should I remove those words in 4.4+?

qinpingli commented on modules/registry-configuring-storage-vsphere.adoc:
```diff
@@ -47,30 +43,7 @@ $ oc get pod -n openshift-image-registry
-----
-exporting *:/mnt/data
-----
+If the storage type is `emptyDIR`, the replica number cannot be greater than `1`.
```
same as baremetal

wzheng1 replied:
This seems correct to me.

bobfuru replied:
Will leave as is, as noted in my previous comment.

qinpingli commented on modules/registry-configuring-storage-vsphere.adoc:
```diff
@@ -130,34 +105,37 @@ $ oc get pv
+  storageClassName: thin
 ----
+
+.. Verify that the PVC was created:
```
In vSphere, OCP supports a default storage class which is the same as other cloudproviders, such as AWS, GCP, so there might be no different from the configuration of AWS EBS.
@wzheng1 Could you help double confirm? Thx!

wzheng1 replied:
Yes, but image registry needs to create pvc manually. Since the default created pvc only supports RWX.

bobfuru replied:
I removed the entire optional step (`Optional: Add a new storage class to a PV`). Correct?

wzheng1 commented:
All in all, Image registry support both RWX and RWO pvc, no matter the PVC is using what kind of storage backend. 
For RWX pvc, Image registry is hardcoded to use RWX+100Gi PVC which can be auto-created if user configure config.imageregistry to spec.pvc.claim(leave the name blank); 
For RWO pvc which is usually used on barematal and vsphere cluster, user has to a create it first and make sure it can be auto-attached to a PV; b. Input the created pvc's name to **spec.pvc.claim** in config.imageregistry, and change spec.rolloutStrategy=**Recreate**;

bobfuru commented:
> For RWO pvc which is usually used on barematal and vsphere cluster, user has to a create it first and make sure it can be auto-attached to a PV; b. Input the created pvc's name to **spec.pvc.claim** in config.imageregistry, and change spec.rolloutStrategy=**Recreate**;

I had already added a procedure for configuring RWO block registry storage for vSphere as part of https://github.com/openshift/openshift-docs/pull/24043/. However, that PR did not add the procedure to the bare metal docs.

I've now updated the bare metal docs to also include the same RWO procedure ("Configuring block registry storage"). Please verify that this is correct, @wzheng1?

bobfuru commented:
@bmcelvee - wanted you to know this PR is currently under review as part of BZ 1850510.

wzheng1 commented on modules/registry-configuring-storage-vsphere.adoc:
```diff
@@ -10,35 +10,31 @@
 ====
-vSphere volumes do not support the `ReadWriteMany` access mode. You must use
-a different storage backend, such as object storage, to configure the registry storage for high-availability.
+{product-title} supports `ReadWriteOnce` access for image registry storage when you have only one replica. To deploy a private image registry, or to configure registry storage for high availability with two or more replicas, you must use a different storage back end that supports `ReadWriteMany`, such as object storage.
```
Same here, please remove "such as object storage"; why to mention "To deploy a private image registry"?

bobfuru replied:
OK, removed "such as object storage" in both modules.

The "private image registry" was wording left in from a previous iteration. Not sure the history behind it, but I agree with your suggestion of instead using "an image registry that supports high availability", which is more clear to me than "private image registry".

wzheng1 commented:
@bobfuru As for "object storage" , maybe we can leave it there, since we found Openshift Container Storage can be used as an object storage. But Openshift QE is not responsible for this part of testing, we cannot ack this for sure.

bobfuru commented:
> @bobfuru As for "object storage" , maybe we can leave it there, since we found Openshift Container Storage can be used as an object storage. But Openshift QE is not responsible for this part of testing, we cannot ack this for sure.

I agree that it's better to leave object storage out of this note, so I will. We do already mention OCS as an example in the prereqs anyway. But as a side note, OCS is storage-agnostic, "supporting file, block, and object storage, either in-house or in hybrid clouds" [1].

[1] https://docs.openshift.com/container-platform/4.3/storage/persistent_storage/persistent-storage-ocs.html

bobfuru commented:
@wzheng1 Thanks for your feedback. I believe this is ready for a final ack from QE, please?

wzheng1 commented:
Acked by QE.

bobfuru commented:
@openshift/team-documentation PTAL

codyhoag commented on modules/installation-registry-storage-block-recreate-rollout-bare-metal.adoc:
```diff
@@ -0,0 +1,32 @@
+
+[IMPORTANT]
+====
+Block storage volumes are supported but not recommended for use with image
```
s/with image registry/with the image registry

codyhoag commented on modules/installation-registry-storage-block-recreate-rollout-bare-metal.adoc:
```diff
@@ -0,0 +1,32 @@
+
+.Procedure
+
+. To set the image registry storage as a block storage type, patch the registry so that it uses the `Recreate` rollout strategy and runs with only `1` replica:
```
Just wanted to check whether it was essential to use the code formatted \`1` over generically referring to it as *one*.

bobfuru replied:
Yeah, I went with the literal value so that it was clear to use the numeral and not the word. I've updated to include both just in case. ¯\_(ツ)_/¯

codyhoag commented on modules/registry-configuring-storage-baremetal.adoc:
```diff
@@ -2,7 +2,7 @@
 // * installing/installing_bare_metal/installing-bare-metal.adoc
 // * installing/installing_bare_metal/installing-restricted-networks-bare-metal.adoc
-// * registry/configuring_registry_storage-baremetal
+// * registry/configuring_registry_storage/configuring-regsitry-storage-baremetal
```
/regsitry/registry

codyhoag reviewed (commented):
Just a few nits; otherwise LGTM!

bobfuru commented:
@huffmanca Do these changes look good to you?

huffmanca commented:
Apologies for the delay.

These changes look good to me!

bobfuru commented:
/cherrypick enterprise-4.6

bobfuru commented:
Opened https://github.com/openshift/openshift-docs/pull/26512 to CP to 4.6, which caused merge conflict due to IBM Z edits. Cc: @vikram-redhat and @ktania

bobfuru commented:
/cherrypick enterprise-4.5

bobfuru commented:
/cherrypick enterprise-4.4
